tests/ops/api/v1/endpoints/test_user_permission_endpoints.py

```diff
@@ -188,62 +189,123 @@ def test_edit_user_permissions(self, db, api_client, generate_auth_header) -> No
+        )
 
-    def test_get_user_permissions_not_authenticated(self, url, api_client):
-        response = api_client.get(url, headers={}, json={})
+    def test_get_user_permissions_not_authenticated(self, api_client, user):
+        response = api_client.get(
+            f"{V1_URL_PREFIX}/user/{user.id}/permission",
+        )
         assert HTTP_401_UNAUTHORIZED == response.status_code
 
-    def test_get_user_permissions_wrong_scope(
-        self, url, api_client, generate_auth_header
-    ):
-        auth_header = generate_auth_header([SAAS_CONFIG_READ])
-        response = api_client.get(url, headers=auth_header, json={})
+    def test_get_user_permissions_wrong_scope(self, db, api_client, user):
+        scopes = [PRIVACY_REQUEST_READ]
+        auth_user = FidesUser.create(
+            db=db,
+            data={"username": "auth_user", "password": "test_password"},
+        )
+        ClientDetail.create_client_and_secret(
+            db,
+            CONFIG.security.oauth_client_id_length_bytes,
+            CONFIG.security.oauth_client_secret_length_bytes,
+            scopes=scopes,
+            user_id=auth_user.id,
+        )
+        auth_header = generate_auth_header_for_user(auth_user, scopes)
+
+        response = api_client.get(
+            f"{V1_URL_PREFIX}/user/{user.id}/permission",
+            headers=auth_header,
+        )
         assert HTTP_403_FORBIDDEN == response.status_code
 
     def test_get_user_permissions_invalid_user_id(
         self, db, api_client, generate_auth_header
     ) -> None:
-        auth_header = generate_auth_header([USER_PERMISSION_READ])
-        invalid_user_id = "bogus_user_id"
-        user = FidesUser.create(
+        scopes = [USER_PERMISSION_READ]
+        auth_user = FidesUser.create(
             db=db,
-            data={"username": "user_1", "password": "test_password"},
+            data={"username": "auth_user", "password": "test_password"},
         )
-
-        permissions = FidesUserPermissions.create(
-            db=db, data={"user_id": user.id, "scopes": [PRIVACY_REQUEST_READ]}
+        ClientDetail.create_client_and_secret(
+            db,
+            CONFIG.security.oauth_client_id_length_bytes,
+            CONFIG.security.oauth_client_secret_length_bytes,
+            scopes=scopes,
+            user_id=auth_user.id,
         )
-        body = {"id": permissions.id, "scopes": [PRIVACY_REQUEST_READ]}
+        auth_header = generate_auth_header_for_user(auth_user, scopes)
+        invalid_user_id = "bogus_user_id"
+
         response = api_client.get(
             f"{V1_URL_PREFIX}/user/{invalid_user_id}/permission",
             headers=auth_header,
-            json=body,
         )
         permissions = FidesUserPermissions.get_by(
             db, field="user_id", value=invalid_user_id
         )
         assert HTTP_404_NOT_FOUND == response.status_code
         assert permissions is None
-        user.delete(db)
```

Comment:
I was wondering that. Does our test suite not clean up tables between tests? This looked like a redundant line to me, because I don't see a lot of manual deletions of records in our tests.

Reply:
AFAIK it doesn't. I could be wrong though. There are other tests in the test suite that are clearing out records that are created during the test. Like this one for example

https://github.com/ethyca/fides/blob/a7a5f971289bd98d2c6248e9c7950d8d18c085b3/tests/ops/api/v1/endpoints/test_email_endpoints.py#L96-L112

My guess is that some tests do it and some don't. It generally won't cause issues but it's a good thing to do to keep tests reliable.

Reply:
Sound like it [does](https://ethyca.slack.com/archives/CBKK8TS74/p1666740056522809?thread_ts=1666732882.446949&cid=CBKK8TS74). Maybe those lines are left over from before the cleanup was implemented? That's pretty batteries-included with most test suites though.

Reply:
That's great. It looks like not everyone, myself included, is aware that it data is automatically cleaned up after each test. 

I looked into the `clear_db_tables` and it uses the `autouse` param. That makes it so all tests use that fixture even if they don't request it. [docs example](https://docs.pytest.org/en/6.2.x/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request)

Maybe we should have a tech debt ticket for cleaning up tests that are manually deleting data? It would be good to have everything be consistent.

Reply:
@TheAndrewJackson for sure open a ticket! We can add this to the testing/quality epic for this quarter